dothething/dothething/ClipsView.swift

```diff
@@ -12,20 +12,19 @@ struct ClipView: View {
     let clip: Clip
 
     var body: some View {
-        // thumbnail of video
-        Image(uiImage: clip.thumbnail)
-            .resizable()
-            .frame(width: 108, height: 192)
-            .cornerRadius(8)
-            .onTapGesture {
-                print("Tapped \(clip.url)")
-                Thinger.playVideo(videoUrl: clip.url)
+        GeometryReader { geometry in
+            if clip.thumbnail == UIImage() {
+                Image("Placeholder")
+                    .resizable()
+            } else {
+                Image(uiImage: clip.thumbnail)
+                    .resizable()
+                    .onTapGesture {
+                        print("Tapped \(clip.url)")
+                        Thinger.playVideo(videoUrl: clip.url)
+                    }
             }
```

Comment:
The changes in this section replace the previous thumbnail display with a conditional rendering based on whether the clip's thumbnail is empty or not. This is a good improvement, as it ensures the app can handle the case where a thumbnail is not available. The code is clear and easy to understand.